### PR TITLE
add local-build cmd 

### PR DESCRIPTION
**Pull request by wayslog**
change makefile to be easy to use .
- [ x ] Make a local lib so that we needn't to install libs into system
- [   ] replace ci-config file by makefile( it's not import , so that you can merge this PR .)

### CODE REVIEW

**Comment by shenli** [comment]
@wayslog Thanks for your contribution! Please sign the CLA.

**Comment by wayslog** [comment]
@shenli  , signed

**Comment by ngaut** [comment]
LGTM

**Comment by ngaut** [comment]
PTAL @siddontang

File: Makefile [hunk 1]
```diff
@@ -7,6 +7,32 @@ all: format build test
 build:
 	cargo build --features ${ENABLE_FEATURES}
 
+local-build: deps-lib build
+	@echo "dependencies build done"
```
Comment (BusyJay) [inline Makefile]: I think this message should be printed before build.

File: Makefile [hunk 2]
```diff
@@ -7,6 +7,32 @@ all: format build test
...
+
+deps-lib: deps-dir
```
Comment (BusyJay) [inline Makefile]: Maybe should depend on dep-rocksdb too.

File: Makefile [hunk 3]
```diff
@@ -7,6 +7,32 @@ all: format build test
...
+	export LD_LIBRARY_PATH="./deps/libs:$LD_LIBRARY_PATH"
```
Comment (BusyJay) [inline Makefile]: I don't think it will work on OS X.

File: Makefile [hunk 4]
```diff
@@ -7,6 +7,32 @@ all: format build test
...
+	export LIBRARY_PATH="$LD_LIBRARY_PATH"
+
+deps-build: deps-dir dep-gflags dep-rocksdb
+
+deps-dir:
```
Comment (BusyJay) [inline Makefile]: Use directory name directly seems better.

File: Makefile (hunk 1 above)
Comment (wayslog) [inline Makefile]: Well, this line should be reported as local-build done .

File: Makefile (hunk 2 above)
Comment (wayslog) [inline Makefile]: You are right.

File: Makefile (hunk 3 above)
Comment (wayslog) [inline Makefile]: But my operation system is openSUSE-42.1 x64 ...

File: Makefile [hunk 5]
```diff
@@ -7,6 +7,32 @@ all: format build test
...
+	mkdir -p deps/libs
+
+dep-gflags: deps-dir
+	wget https://github.com/gflags/gflags/archive/v2.1.2.tar.gz -O deps/gflags.tar.gz
+	tar -zxf deps/gflags.tar.gz -C deps
+	cd deps/gflags-2.1.2 && cmake . && make && cp lib/* ../libs
```
Comment (siddontang) [inline Makefile]: can this work in Mac OS X?

**Comment by ngaut** [comment]
@wayslog Ping

**Comment by wayslog** [comment]
@ngaut ，pong . Sorry , I was a little busy now . After get everything gone , I will fix it .

**Pull request closed without merging**